hack/kind-install-for-capd.sh

```diff
@@ -60,15 +67,33 @@ nodes:
       containerPath: /var/run/docker.sock
 containerdConfigPatches:
 - |-
-  [plugins."io.containerd.grpc.v1.cri".registry.mirrors."localhost:${reg_port}"]
-    endpoint = ["http://${reg_name}:${reg_port}"]
+  [plugins."io.containerd.grpc.v1.cri".registry]
+    config_path = "/etc/containerd/certs.d"
+EOF
+
+# 4. Add the registry config to the nodes
+#
+# This is necessary because localhost resolves to loopback addresses that are
+# network-namespace local.
+# In other words: localhost in the container is not localhost on the host.
+#
+# We want a consistent name that works from both ends, so we tell containerd to
+# alias localhost:${reg_port} to the registry container when pulling images
+REGISTRY_DIR="/etc/containerd/certs.d/localhost:${reg_port}"
+for node in $(kind get nodes --name "$KIND_CLUSTER_NAME"); do
```

Comment:
This only works for the kind node we just created, right?

So the registry is only available on this one mgmt/bootstrap cluster node. But I assume this is not a problem and was always the case? (i.e. wl clusters created by CAPD couldn't use the registry)

Reply:
Yeah - we never really enabled the registry on workload clusters, though it would be possible. This change unfortunately makes it much more difficult to enable in CAPD however.

Reply:
https://github.com/kubernetes-sigs/cluster-api/issues/8736